falcon/request.py

```diff
@@ -371,20 +373,20 @@ def if_unmodified_since(self):
     def range(self):
         try:
             value = self.env['HTTP_RANGE']
-            if value.startswith('bytes='):
-                value = value[6:]
+            if '=' in value:
+                unit, sep, range = value.partition('=')
```

Comment:
Ah, unfortunately this will end up masking the builtin `range` function, so that variable will need a different name.